jellyfin.tf

```diff
@@ -202,85 +222,44 @@ resource "aws_instance" "jellyfin_server" {
     ]
   }
 
+  // Creates nginx conf file for jellyfin server based on ALB or Domain name
   provisioner "file" {
-    content = <<EOF
-server {
-  listen 80;
-    server_name ${var.HOSTED_ZONE_ID == "" ? "~^${aws_lb.jellyfin_alb.name}.*\\.elb\\.amazonaws.com$" : trimsuffix(data.aws_route53_zone.jellyfin_domain.0.name,".") };
-  location / {
-    # Proxy main Jellyfin traffic
-    proxy_pass http://localhost:8096/;
-    proxy_set_header Host $host;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto $scheme;
-    proxy_set_header X-Forwarded-Protocol $scheme;
-    proxy_set_header X-Forwarded-Host $http_host;
-
-    # Disable buffering when the nginx proxy gets very resource heavy upon streaming
-    proxy_buffering off;
-  }
-  location /socket {
-    # Proxy Jellyfin Websockets traffic
-    proxy_pass http://localhost:8096/socket;
-    proxy_http_version 1.1;
-    proxy_set_header Upgrade $http_upgrade;
-    proxy_set_header Connection "upgrade";
-    proxy_set_header Host $host;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_set_header X-Forwarded-Proto $scheme;
-    proxy_set_header X-Forwarded-Protocol $scheme;
-    proxy_set_header X-Forwarded-Host $http_host;
-  }
-}
-EOF
+    content = templatefile("templates/server.conf.tmpl", { server_name = local.server_name })
     destination = "/tmp/instant-jellyfin.conf"
   }
 
+  // Lays down cron for syncing files form s3 bucket to local storage
   provisioner "file" {
-    content = <<EOF
-#!/bin/bash
-echo -e "$(crontab -l 2>/dev/null | grep -v jellyfin-s3-sync)\n*/5 * * * * /bin/bash ~/jellyfin/scripts/s3sync.sh #jellyfin-s3-sync" | crontab -
-EOF
+    content = file("files/start-sync.sh") 
     destination = "/tmp/start-s3sync.sh"
   }
 
+  // Creates cron job script with bucket variable
   provisioner "file" {
-    content = <<EOF
-#!/bin/bash
-aws s3 sync s3://${aws_s3_bucket.jellyfin_media.id} ~/jellyfin/media --delete
-EOF
+    content = templatefile("templates/s3sync.sh.tmpl", { BUCKET = aws_s3_bucket.jellyfin_media.id })
     destination = "/tmp/s3sync.sh"
   }
 
+  // Formats media EBS Volume and mounts to ~/jellyfin/media
+  provisioner "remote-exec" {
+    inline = [
+      var.EBS_MEDIA_VOLUME_SIZE == "" ? "" : "sudo mkfs -t xfs /dev/nvme1n1 && sudo mount ${local.EBS_Device} ~/jellyfin/media"
```

Comment:
Before this is merged, I'd like to make sure the mount is persistent even on reboot. See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-using-volumes.html, which has details on how to do this. I'll work on a little bit of scripting for this.

Reply:
Added a couple lines to do it, will need to test that they work. on a new build.

Reply:
Terrific, I'm testing this now.

Reply:
Just finished testing this, see my new comments for how to make this work.